loopy/check.py

```diff
@@ -933,6 +954,11 @@ def pre_schedule_checks(kernel, callables_table):
             check_for_integer_subscript_indices(kernel, callables_table)
 
         check_functions_are_resolved(kernel)
+        # Ordering restriction:
+        # check_sub_array_ref_inames_not_within_or_redn_inames should be done
+        # before check_bounds. check_bounds involves certain assertions
+        # triggering this restriction.
```

Comment:
```suggestion
        # before check_bounds. check_bounds involves certain assumptions
        # requiring this restriction.
```
(The phrasing is a bit opaque. Sharpen?)

Reply:
Was more implementation specific so just x-ref-ed it: 3eb732c.